src/game_state/game.py

```diff
@@ -30,23 +36,50 @@ def __init__(self):
             )
         )
         self.score = 0
-        self.fruit_spawn = True
 
-    def touch_boundaries(self) -> bool:
+    def _touch_boundaries(self) -> bool:
+        """Checks whether the snake's head is touching the
+        boundaries of the board.
+
+        Returns:
+            bool: True in case the snake's head is touching the boundaries, False otherwise.
+        """
         return (
             self.snake.position[0] < 0
             or self.snake.position[0] > self.window_x - 10
             or self.snake.position[1] < 0
             or self.snake.position[1] > self.window_y - 10
         )
 
-    def touch_body(self) -> bool:
+    def _touch_body(self) -> bool:
+        """Checks wheter the snake's head is touching its body.
+
+        Returns:
+            bool: True in case the snake's head is touching its body, False otherwise.
+        """
         return any(self.snake.position == block for block in self.snake.body[1:])
 
     def game_over(self):
-        return self.touch_boundaries() or self.touch_body()
+        """Checks for the conditions of Game Over.
+
+        The conditions are:
+            * Snake's head touches boundaries.
+            * Snake's head touches body.
+
+        Returns:
+            bool: True if any condition is met, False otherwise.
```

Comment:
We should add the condition that the snake occupies the whole board. This rarely happens but if we later create an automatic snake that plays perfectly it should achieve it.

Reply:
Ack